README.md

```diff
@@ -2,20 +2,118 @@
 
 > Test client for Xo-Server
 
-## Usage
-
-xo-server needs to be installed on `localhost:9000`.
-You should also have a `admin@admin.net` account with password `admin`.
+## Adding a test
 
-There should be a Xen Server available at `192.168.1.3`with the credentials `root` and password `qwerty`.
+### Organization
 
-Warning ! This client will create and delete a lot of items (users, groups, servers ...) to test features
+```
+src
+├─ user
+|   ├─ __snapshots__
+|   |     └─ index.spec.js.snap 
+|   └─ index.spec.js
+├─ job
+¦   └─ index.spec.js
+¦
+¦
+├─ _xoWithTestHelpers.js
+└─ util.js
+```
 
+The tests can describe xo methods or scenarios:  
 ```
-> npm ci
-> npm test
+describe("user", () => {
+
+  // testing a method
+  describe(".set()", () => {
+    it("sets an email", async () => {
+      // some tests using xo methods and helpers from _xoWithTestHelpers.js
+      id = await xo.createUser(SIMPLE_USER);
+      expect(await xo.call("user.set", params)).toBe(true);
+      expect(omit(await xo.getUser(id), "id")).toMatchSnapshot();
+    });
+  });
+      
+  // testing a scenario
+  test("create two users, modify a user email to be the same with the other and fail trying to connect them", () => {
+    /* some tests */
+  });
+
+});
 ```
 
+### Best practices
+
+- The test environment must remain the same before and after each test:  
+  * each resource created must be deleted  
+  * existing resources should not be altered
+
+- Make a sentence for the title of the test. It must be clear and consistent.
+
+- If the feature you want to test is not implemented : write it and skip it, using `it.skip()`.
+
+- Take values ​​that cover the maximum of testing possibilities.
+
+- If you make tests which keep track of large object, it is better to use snapshots.  
+
+- `_xoWithTestHelpers.js` contains helpers to create temporary resources and to interface with XO.  
+  You can use it if you need to create resources which will be automatically deleted after the test:  
+  ```
+  import xo from "util"
+
+  describe(".create()", () => {
+    it("creates a user without permission", async () => {
+      // The user will be deleted automatically at the end of the test
+      const userId = await xo.createUser({
+        email: "wayne1@vates.fr",
+        password: "batman1",
+      });
+      expect(omit(await xo.getUser(userId), "id")).toMatchSnapshot();
+    });
+  });
+  ```
+
+  The available helpers:  
+  * `createUser(params)`  
+  * `deleteAllUsers()`  
+  * `getUser(id)`  
+  * `createTempJob(jobId)`  
+  * `deleteTempJobs()`  
+  * `createTempBackupNgJob(params)`  
+  * `deleteTempBackupNgJob()`  
```

Comment:
You can remove the `delete*` methods, they will be automatically invoked, that's the point of the helpers :slightly_smiling_face: